pkg/cli/admin/upgrade/recommend/recommend.go

```diff
@@ -142,6 +149,36 @@ func (o *options) Run(ctx context.Context) error {
 		fmt.Fprintf(o.Out, "info: An update is in progress.  You may wish to let this update complete before requesting a new update.\n  %s\n", strings.ReplaceAll(c.Message, "\n", "\n  "))
 	}
 
+	if o.precheckEnabled {
+		conditions, err := o.precheck(ctx)
+		if err != nil {
+			fmt.Fprintf(o.Out, "Failed to check for at least some preconditions: %v\n", err)
+		}
+		var happyConditions []string
+		var unhappyConditions []string
+		for _, condition := range conditions {
+			if condition.Status == metav1.ConditionTrue {
+				happyConditions = append(happyConditions, fmt.Sprintf("%s (%s)", condition.Type, condition.Reason))
+			} else {
+				unhappyConditions = append(unhappyConditions, condition.Type)
```

Comment:
Why omit reason for unhappy ones? I'd expect reasons to be more interesting for unhappy ones even.
EDIT: Ah okay, I get it now, details for unhappy ones are treated separately L174+